src/uTensor/ops/Transpose.hpp

```diff
@@ -0,0 +1,130 @@
+#ifndef UTENSOR_TRANSPOSE_H
+#define UTENSOR_TRANSPOSE_H
+
+#include "context.hpp"
+#include "types.hpp"
+#include "tensor.hpp"
+#include "uTensor_util.hpp"
+#include "operatorBase.hpp"
+
+#include <cstring>
+
+namespace uTensor {
+namespace ReferenceOperators {
+
+// Transpose (Swap Axes) as a port from Numpy
+// using stride interation in the order of transpose axes
+template <typename Tin>
+class TransposeOperator : public OperatorInterface<1, 1> {
+/* reshape input as the shape of output*/
+public:
+  TransposeOperator(const TensorShape&& axes) : _axes(axes) {}
+  TransposeOperator(const TensorShape& axes) : _axes(axes) {}
+  
+  enum names_in : uint8_t { input };
+  enum names_out : uint8_t { output };
+
+  virtual void compute(){
+    Tensor& input_tensor = inputs[input].tensor();
+    TensorShape& input_shape = input_tensor.get_shape();
+    input_shape.update_dims();
+
+    // Strides are used to iterate over the dataset, and transfer
+    // the input tensor data, into the output tensor
+    TensorStrides* input_strides = new TensorStrides(input_shape);
+
+    Tensor& output_tensor = outputs[output].tensor();
+
+    // Create a placeholder to calculate the output shape
+    TensorShape* output_shape = new TensorShape(1,1,1,1);
```

Comment:
```suggestion
    // Normally this would reference output shape, but since this could (usually would) be referencing the input, let's keep a dedicated value
    TensorShape output_shape = TensorShape(1,1,1,1);
```